File: tests/xcalablemp/global-view/task/F/nocomm.f90
!$xmp nodes p(4,4)

!$xmp template t(100,100)
!$xmp distribute t(block,cyclic(5)) onto p

  integer k

  k = 0

!$xmp task on p(2:3,1:4:2)
  k = 1
!$xmp end task

!$xmp task on p(2:3,1:4:2) nocomm
  k = 0
!$xmp end task

!$xmp reduction (+:k)

!$xmp task on p(1,1)
  if (k /= 0) then
     write(*,*) "ERROR"
     call exit(1)
  end if
!$xmp end task

  k = 0

!$xmp task on t(1,1)
  k = 1
!$xmp end task

!$xmp task on t(1,1) nocomm
  k = 0
!$xmp end task

!$xmp reduction (+:k)

!$xmp task on p(1,1)
  if (k /= 0) then
     write(*,*) "ERROR"
     call exit(1)
  else
     write(*,*) "PASS"
  end if
!$xmp end task

end program

File: tests/XMP/global-view/task/C/nocomm.c
#include <stdlib.h>
#include <stdio.h>

#pragma xmp nodes p[4][4]

#pragma xmp template t[1:100][1:100]
#pragma xmp distribute t[cyclic(5)][block] onto p

int main(void){

  int k;

  k = 0;

#pragma xmp task on p[0:4:2][1:2]
  {
    k = 1;
  }

#pragma xmp task on p[0:4:2][1:2] nocomm
  {
    k = 0;
  }

#pragma xmp reduction (+:k)

#pragma xmp task on p[0][0]
  {
    if (k != 0){
      printf("ERROR\n");
      exit(1);
    }
  }
  
  k = 0;

#pragma xmp task on t[1][1]
  {
    k = 1;
  }

#pragma xmp task on t[1][1] nocomm
  {
    k = 0;
  }

#pragma xmp reduction (+:k)

#pragma xmp task on p[0][0]
  {
    if (k != 0){
      printf("ERROR\n");
      exit(1);
    }
    else {
      printf("PASS\n");
    }
  }

  return 0;

}
